script.js

```diff
@@ -1,4 +1,260 @@
+userName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+  if (currentGameMode == "waiting for user name...") {
+    console.log(currentGameMode);
+    userName = input;
+    console.log("user name: " + userName);
+    currentGameMode = "normal or reverse...";
+    myOutputValue = "Please choose between 'normal' or 'reverse'.";
+    return myOutputValue;
+  } else if (currentGameMode == "normal or reverse...") {
+    console.log(currentGameMode);
+    currentGameMode = input;
+    myOutputValue = "Please choose between 'scissors', 'paper' or 'stone'.";
+    return myOutputValue;
+  } else if (currentGameMode == "normal") {
+    myOutputValue =
+      "Please try again. Use only 'scissors', 'paper' or 'stone'.";
+    if (input == "scissors" || input == "paper" || input == "stone") {
+      myOutputValue = playSPS(input);
+    }
+    return myOutputValue;
+  } else if (currentGameMode == "reverse") {
+    console.log(currentGameMode);
+    myOutputValue =
+      "Please try again. Use only 'scissors', 'paper' or 'stone'.";
+    if (input == "scissors" || input == "paper" || input == "stone") {
+      myOutputValue = playSPS(input);
+    }
+    return myOutputValue;
+  }
+};
+
+var playSPS = function (input) {
+  // roll a random computer hand
+  computerHand = rollHand();
+  // input is player hand
+  playerHand = input;
+  return rulesSPS(playerHand, computerHand);
+};
+
+var rulesSPS = function (playerHand, computerHand) {
+  if (playerHand == computerHand) {
+    console.log("player hand: " + playerHand);
+    console.log("computer hand: " + computerHand);
+    drawCount = drawCount + 1;
+    console.log("draws: " + drawCount);
+    console.log("player wins: " + playerWinCount);
+    console.log("computer wins: " + computerWinCount);
+    return responseSPS("draw", playerHand, computerHand);
+  } else if (
+    (playerHand == "scissors" && computerHand == "paper") ||
+    (playerHand == "paper" && computerHand == "stone") ||
+    (playerHand == "stone" && computerHand == "scissors")
+  ) {
+    console.log("player hand: " + playerHand);
+    console.log("computer hand: " + computerHand);
+    if (currentGameMode == "normal") {
+      playerWinCount = playerWinCount + 1;
+    } else if (currentGameMode == "reverse") {
+      computerWinCount = computerWinCount + 1;
+    }
+    console.log("draws: " + drawCount);
+    console.log("player wins: " + playerWinCount);
+    console.log("computer wins: " + computerWinCount);
+    if (currentGameMode == "normal") {
+      return responseSPS("win", playerHand, computerHand);
+    } else if (currentGameMode == "reverse") {
+      return responseSPS("lose", playerHand, computerHand);
+    }
+  } else if (
+    (computerHand == "scissors" && playerHand == "paper") ||
+    (computerHand == "paper" && playerHand == "stone") ||
+    (computerHand == "stone" && playerHand == "scissors")
```

Comment:
good to have, but not need to have: there are only 3 possible outcomes (let's assume normal mode), (1) draw, (2) playerwin, (3) compwin. So if we have checked for draw, then checked for playerwin, the program will only reach line 76 in the event of a draw, so this conditional will always evaluate to true whenever the program reaches here. does that make sense?